src/components/localtunneldebugger/localtunneldebugger.ts

```diff
@@ -0,0 +1,46 @@
+import * as vscode from 'vscode';
+
+import { LocalTunnelDebugProvider } from './localtunneldebugger.extension';
+import { getLocalTunnelDebugProvider } from '../config/config';
+
+export class LocalTunnelDebugger {
+    private readonly providers = Array.of<LocalTunnelDebugProvider>();
+
+    register(provider: LocalTunnelDebugProvider): void {
+        console.log(`Registered local tunnel debugger type ${provider.id}`);
+        this.providers.push(provider);
+    }
+
+    startLocalTunnelDebugProvider(target?: any): void {
```

Comment:
💡 Idea\thought share: possible thought process behind below is: bail early if none of the conditions satisfy.

Given below 3 major if conditions, if:

* `providers.length` is `non-empty` and 
* `providename` if `non-empty` and 
*  `providerToUse` is `not undefined`, there is nothing which probably gets run?

In that case should we do all the existence check upfront and then bail with some message that you have got all necessary stuff and nothing actually happened?

Reply:
* Minor idea addition: we could compartmentalize these conditions into its own small `functs` units, and call it from main function as well.

Reply:
@Tatsinnit  thank you so much for adding these comments! Dan told me you had messaged him and mentioned forking this PR. Perhaps we can sync up later today? I've just addressed Dan's feedback, hopefully it won't cause any conflicts. :)

Reply:
Thank you so much @amsoedal, Yeah I dint wanted to just comment :) was planning to possibly add a PR to your Fork and see if that helps. Yes, please feel free to ping me internally in teams please. I think Dan will have my alias, since its git I will not ask your internal alias here.

Reply:
@Tatsinnit I merged the PR, will resolve any comments I think were addressed by it, but please feel free to reopen them or open new ones if you have more thoughts. I had to tweak a couple things (the version of tslint used in the CI build was unhappy about the use of "??", and I updated the error conditions as the latter message was always getting used). Otherwise I definitely think the code looks cleaner, really appreciate the help on that!